pandas/core/resample.py

```diff
@@ -1927,6 +1927,9 @@ def get_resampler_for_grouping(
     """
     # .resample uses 'on' similar to how .groupby uses 'key'
     tg = TimeGrouper(freq=rule, key=on, **kwargs)
+    # GH 59350: Index should be ignored when using the on keyword argument to resample
+    if on is not None:
+        groupby.obj = groupby.obj.reset_index(drop=True)
```

Comment:
I'm concerned about modifying the state of `groupby.obj` here. For example, we haven't filled the groupby caches at this point, so what if the `"group"` column from the linked issue is in the index? I tried this on main and it raises but I don't think it should:

```python
df = pd.DataFrame(dict(
    datetime=[pd.to_datetime('2024-07-30T00:00Z'), pd.to_datetime('2024-07-30T00:01Z')],
    group=['A', 'A'],
    value=[100, 200],
)).set_index("group")
result = df.groupby('group').resample('1min', on='datetime').aggregate(dict(value='sum'))
print(result)
```

If this did work on main, then I think it certainly wouldn't work given this PR.

Reply:
Actually this PR solves the error on main (`ValueError: invalid literal for int() with base 10: 'A'`) and gives the correct result. I added a test case to illustrate

Reply:
Thanks for checking - I forgot we do store the grouping data in the corresponding groupings objects upon construction. In the following example, if you uncomment the `resample` line, you get the correct index. However with that line you get a RangeIndex.

```python
df = pd.DataFrame(dict(
    datetime=[pd.to_datetime('2024-07-30T00:00Z'), pd.to_datetime('2024-07-30T00:01Z')],
    group=['A', 'A'],
    value=[100, 200],
)).set_index("group")
gb = df.groupby('group')
result = gb.resample('1min', on='datetime').aggregate(dict(value='sum'))
result = gb[["value"]].transform("sum")
print(result)
```